lang/src/org/partiql/lang/eval/LikeMatchingAutomata.kt

```diff
@@ -401,8 +412,14 @@ private fun buildDFA(dfaAlphabet: Set<Alphabet>,
         val deltaUpdates: List<Pair<Pair<Set<NFAState>, Alphabet>, Set<NFAState>>> =
             dfaAlphabet.map {
                 Pair(Pair(nfaStates, it),
-                    nfaStates.fold(setOf<NFAState>()) { acc, state ->
-                        acc.union(state.getOutgoingStates(it))
+                    HashSet<NFAState>().apply {
+                        nfaStates.forEach { state ->
+                            addAll(state.getOutgoingStates(it))
+
+                            if (Thread.interrupted()) {
+                                throw InterruptedException()
+                            }
+                        }
```

Comment:
This is the change in this PR that results in the biggest performance improvement.  The other stuff is incremental.